src/Microsoft.AspNet.Mvc.Core/Filters/FormatFilterAttribute.cs

```diff
@@ -0,0 +1,139 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.AspNet.Mvc.Description;
+using Microsoft.Framework.DependencyInjection;
+using Microsoft.Framework.OptionsModel;
+using Microsoft.Net.Http.Headers;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// This will look at the format parameter if present in the route data or query data and sets the content type in 
+    /// ObjectResult corresponding to the format value.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = false, Inherited = true)]
+    public class FormatFilterAttribute : Attribute, IFormatFilter, IResourceFilter, IResultFilter
+    {
+        /// <summary>
+        /// As a resourceFilter, this filter looks at the request and rejects it
+        /// before going ahead if
+        /// 1. The format in the request doesnt match any format in the map.
+        /// 2. If there is a conflicting producesFilter.
+        /// </summary>
+        /// <param name="context"></param>
+        public void OnResourceExecuting([NotNull] ResourceExecutingContext context)
+        {
+            var format = GetFormat(context);
+
+            if (!string.IsNullOrEmpty(format))
+            {
+                var formatContentType = GetContentType(format, context);
+                if (formatContentType == null)
+                {
+                    // no contentType exists for the format, return 404
+                    context.Result = new HttpNotFoundResult();
+                }
+                else
+                {
+                    var responseTypeFilters = context.Filters.OfType<IApiResponseMetadataProvider>();
+                    if (responseTypeFilters.Count() != 0)
+                    {
+                        var contentTypes = new List<MediaTypeHeaderValue>();
+
+                        foreach (var filter in responseTypeFilters)
+                        {
+                            filter.SetContentTypes(contentTypes);
+                        }
+
+                        if (!contentTypes.Any(c => c.IsSubsetOf(formatContentType)))
```

Comment:
My comment about this code seems to have disappeared, still relevant.

> `responseTypeFilters.Count() != 0` is not useful, remove it.

Your concern is that `contentTypes.Any(c => c.IsSubsetOf(formatContentType))` returns `false` when `contentTypes` is empty. You're correct, and that's still possible when you have `responseTypeFilters > 1` so handle that case because it's the one you really should be concerned about.